src/vs/platform/request/node/proxy.ts

```diff
@@ -44,7 +47,8 @@ export async function getProxyAgent(rawRequestURL: string, env: typeof process.e
 		rejectUnauthorized: isBoolean(options.strictSSL) ? options.strictSSL : true,
 	};
 
+	// {{SQL CARBON EDIT}} Using updated 'http-proxy-agent' and 'https-proxy-agent' packages.
 	return requestURL.protocol === 'http:'
-		? new (await import('http-proxy-agent'))(opts as any as Url)
-		: new (await import('https-proxy-agent'))(opts);
+		? new HttpProxyAgent(opts)
```

Comment:
Is this something that affects VS Code then? If so, we should open a PR for them as well (and then ideally cherry pick their commit over here)

Reply:
Yes it does, and that's the possible cause of https://github.com/OfficeDev/TeamsFx/issues/7995.

I can contribute a PR for the same thing there, but I'm not sure how soon they will take it in. In our case it's a blocker and we need it soon for upcoming release. I'd still contribute and remove these `SQL Carbon Edit` tags later when VS Code gets updated.

Reply:
Yup, that sounds great! (then if they get it in we can update our code with whatever they end up having if it's different for some reason to just keep things in sync)